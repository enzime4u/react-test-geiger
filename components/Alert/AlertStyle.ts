import styled from "styled-components";

export const AlertContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  padding: 10;
  background-color: ${({ theme }) => theme.colors.alert};
  position: absolute;
  height: 80px;
  top: 0;
  left: 0;
  right: 0;
`;
