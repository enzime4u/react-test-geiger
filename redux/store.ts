import { configureStore } from '@reduxjs/toolkit';
import formSlice from './formSlice';
import alertSlice from './alertSlice';

const store = configureStore({
  reducer: {
    newsLetter: formSlice,
    alert: alertSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;

export default store;
