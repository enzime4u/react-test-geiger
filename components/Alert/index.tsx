import React from 'react';
import AlertComponent from './AlertStructure';
import { useSelector } from 'react-redux';
import { RootState } from '@/redux/store';

const Alert = () => {
  const { message, isVisible } = useSelector((state: RootState) => state.alert);

  const componentProps = {
    message,
    isVisible,
  };

  return (
    <div style={{ display: isVisible ? 'block' : 'none' }}>
      <AlertComponent {...componentProps} />
    </div>
  );
};

export default Alert;
