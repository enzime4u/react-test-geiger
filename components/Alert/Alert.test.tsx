import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import { ThemeProvider } from 'styled-components';
import { theme } from '@/configs';
import Alert from '.';

const withProviders = (children: React.JSX.Element) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

it('it should match the snapshot and render', async () => {
  const { container } = render(withProviders(<Alert />));

  expect(container).toMatchSnapshot();
  expect(container).toBeInTheDocument();
});
