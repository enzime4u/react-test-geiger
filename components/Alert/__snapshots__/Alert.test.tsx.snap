// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot and render 1`] = `
<div>
  <div
    style="display: none;"
  >
    <div
      class="sc-4fde7620-0 ZYYvZ"
    >
      <h1 />
    </div>
  </div>
</div>
`;
